src/kOS.Safe/Execution/Stack.cs

```diff
@@ -168,31 +215,19 @@ public string Dump()
             try
             {
                 var builder = new StringBuilder();
-                builder.AppendLine("Stack dump: stackPointer = " + stackPointer);
+                builder.AppendLine("Stack dump: count = " + count + ", aboveCount = " + aboveCount);
 
-                // Print in reverse order so the top of the stack is on top of the printout:
-                // (actually given the double nature of the stack, one of the two sub-stacks
-                // inside it will always be backwardly printed):
-                for (int index = stack.Count - 1; index >= 0; --index)
+                // To simulate a single stack with the SP in the middle, we go through the aboveStack bottom up, then
+                // the normal stack top down
```

Comment:
If we're changing the underlying implementation, I don't mind exposing that fact to users in the `Dump()` output.  This is meant to be a dump of what is really going on under the hood, so we should just print out the two stacks as two different stacks, labeled accordingly.